### DISCUSSION TITLE
Form control - required

### DISCUSSION THREAD

**Discussion by richware** [twbs/bootstrap#35929] [Ideas]
I did not see anywhere that the 'required' attribute was handled, therefore I have created the followng SASS:

`.form-control
  &:not([type=file])
    &:required
      @include auto-duo-gradient($form-required-bg,5%)`

Which generates the following:

`.form-control:not([type=file]):required {
  background: linear-gradient(#fff3f5, #FFD9E0) !important;
  color: #000000 !important;
}
`

Regards,
Rich

**Comment by WinterSilence** [comment]
you miss `.form-select`, `.form-range`:
~~~scss
.form-control,
.form-select,
.form-range {
   &:required {
      ...
      .disabled&,
      [disabled]& {
         ...
      }
  }
}
~~~

**Reply by richware** [reply]
So will this be included in the next version of bootstrap?

**Reply by WinterSilence** [reply]
@richware if you don't send PR - no)
